Source/OxyPlot/Axes/Axis.cs

```diff
@@ -1402,6 +1470,33 @@ internal virtual void UpdateTransform(OxyRect bounds)
             }
 
             this.SetTransform(newScale, newOffset);
+
+            if (this.MinimumDataMargin > 0)
+            {
+                this.ClipMinimum = this.InverseTransform(a0 - (this.MinimumDataMargin * marginSign));
+            }
+            else
+            {
+                this.ClipMinimum = this.ActualMinimum;
+            }
+
+            if (this.MaximumDataMargin > 0)
+            {
+                this.ClipMaximum = this.InverseTransform(a1 + (this.MaximumDataMargin * marginSign));
+            }
+            else
+            {
+                this.ClipMaximum = this.ActualMaximum;
+            }
+
+            this.ActualMaximumAndMinimumChangedOverride();
+        }
+
+        /// <summary>
+        /// Invoked when <see cref="ClipMinimum"/> and <see cref="ClipMaximum"/> are changed.
+        /// </summary>
+        protected virtual void ActualMaximumAndMinimumChangedOverride()
```

Comment:
... this is technically true; the `Actual` bounds are changed a lot without anyone knowing, but the method name is preserved. I'll change it to "Invoked when <see cref="ActualMinimum"/>, <see cref="ActualMaximum"/>, <see cref="ClipMinimum"/>, and <see cref="ClipMaximum"/> are changed."